pkg/policy/rule.go

```diff
@@ -85,7 +85,7 @@ func mergeL4IngressPort(ctx *SearchContext, endpoints []api.EndpointSelector, en
 		if ep, ok := existingFilter.L7RulesPerEp[hash]; ok {
 			switch {
 			case len(newL7Rules.HTTP) > 0:
-				if len(ep.Kafka) > 0 {
+				if len(ep.Kafka) > 0 || len(ep.L7) > 0 {
```

Comment:
This does not seem to detect a conflict if l7proto is set but the l7 map is empty. I would check for `ep.L7Proto != ""`.

Reply:
Right, in the same vain I'm changing the (new) case to test for the `L7Proto` rather than an non-empty `L7`